src/awscr-s3/client.cr

```diff
@@ -105,7 +105,7 @@ module Awscr::S3
     end
 
     private def http
-      Http.new(signer, @region)
+      @http ||= Http.new(signer, @region, @endpoint)
```

Comment:
I don't see a reason why this shouldn't be memoized, especially because you'd usually be using the same client object for making multiple requests.

Reply:
I don't know why this is the case, but if this is memoized when `FileUploader` takes the mutlipart route errors occur. Let's remove the memo and if we think we want to memo this investigate that bug in another PR :) @lipanski